src/main/java/calculator/service/CalculatorService.java

```diff
@@ -0,0 +1,33 @@
+package calculator.service;
+
+import calculator.domain.Calculator;
+import calculator.domain.PostFixCalculator;
+import calculator.exception.NotSaveException;
+import calculator.repository.CalculatorRepository;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CalculatorService {
+    private CalculatorRepository calculatorRepository;
+    public CalculatorService(CalculatorRepository calculatorRepository) {
+        this.calculatorRepository = calculatorRepository;
+    }
```

Comment:
```suggestion
    private final CalculatorRepository calculatorRepository;
    public CalculatorService(CalculatorRepository calculatorRepository) {
        this.calculatorRepository = calculatorRepository;
    }
```
생성자를 통해서만 객체가 초기화된다면 `final`을 붙여주면 어떨까요? 
`final`을 붙이면 어떤 이점을 가져올 수 있을까요?

Reply:
final을 통해서 필드변수를 초기화 하게 되면, 초기화 이후에 새로운 값을 다시 설정할 수 없게 됩니다.
`CalculatorRepository`의 경우 저장소가 한번 생성된 이후 새로운 저장소를 생기게 하면 기존에 있던 값을 삭제할 수 있으니까 붙이는 게 좋을 거 같네요!

그렇다면 `CalculatorService`나 `CalculatorController`의 경우에도 한 번 생성한 후 프로그램 실행 도중에 변경할 필요가 없으니까 붙이는 게 좋을까요?

Reply:
넵 변경할 필요가 없으면 `final`을 습관적으로 붙여 불변을 만들어주는 습관이 좋습니다.
불변을 유지시켜 주면 외부에서 사용할 때 좀 더 안전하게 사용할 수 있겠죠?